eta/core/datasets.py

```diff
@@ -2751,20 +2751,25 @@ class SchemaFilter(DatasetTransformer):
     is None, no filtering is done.
     '''
 
-    def __init__(
-            self, schema, remove_objects_without_attrs=False,
-            prune_empty=True):
-        '''Initialize the SchemaFilter with a schema.
+    def __init__(self, schema, remove_objects_without_attrs=False,
+                 object_labels_to_filter=None, prune_empty=True):
+        '''Initialize the SchemaFilter
 
         Args:
             schema: a VideoLabelsSchema or ImageLabelsSchema
             remove_objects_without_attrs: whether to remove objects with no
-                attributes, after filtering. By default, this is False
+                attributes, after filtering. Use the `object_labels_to_filter`
+                argument to control which object labels are filtered. By
+                default, this is False
+            object_labels_to_filter: an optional list of object labels for
+                which to restrict attention when filtering objects with no
+                attributes. If None, all objects are filtered
```

Comment:
One other thing related to style- I see that you sometimes introduce docstrings that use the `parameter (type): description` syntax to document the type of the argument. I wish that all of ETA did that, as that would allow us to use Sphinx (as we do in https://github.com/voxel51/api-py) to generate auto-documentation that can parse out parameter types. However, we haven't adopted that in the existing codebase, so the style guide says to follow the existing docstring convention in the repo.

tl;dr: no `(type)` for now